lib/XMLParser.php

```diff
@@ -97,9 +99,13 @@ public function parse($xml)
 
                 case \XMLReader::SIGNIFICANT_WHITESPACE:
                     if ($this->whitespaceCallback) {
-                        call_user_func($this->whitespaceCallback);
+                        call_user_func($this->whitespaceCallback, $this);
                     }
                     break;
+
+                case \XMLReader::COMMENT:
+                    // Don't do anything with comments
+                    break;
```

Comment:
If we don't care about comments, why not leave this part all together? It has no meaning here, doesn't it?

Reply:
The reason I added it was because I added a `default:` to the switch statement to see if there was anything getting parsed that I was missing. I added the comment case to hide it from the output.

I think it is quite useful as an enumeration of things to expect when parsing XML but equally happy to remove it again.